### PR TITLE
chore: mask registry creds in webui [DET-7013]

### PR DESCRIPTION
**Pull request by trentwatt**
## Description

chore: mask registry creds in experiment config/fork [DET-7013]

## Test Plan

- have an experiment
- click fork, add
```
registry_auth:
    username: nameduser
    password: wordpass
``` 
to the `environment` section of experiment config
- make fork
- go to configuration page of the new experiment
- `registry_auth` should not be there
- pause and continue experiment, expand continue experiment dialog
- `registry_auth` should not be there
- confirm continue, previous auth should be sent in request
- click fork again and expand experiment dialog
- `registry_auth` should not be there
- confirm fork, previous auth should be sent in request
- on new experiment, click fork again
- supply new `registry_auth`
- new auth should be sent in request

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DET-7013]: https://determinedai.atlassian.net/browse/DET-7013?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/hooks/useModal/useModalExperimentCreate.tsx [hunk 1]
```diff
@@ -215,15 +221,38 @@ const useModalExperimentCreate = (props?: Props): ModalHooks => {
...
+      if(!userConfig?.environment?.registry_auth && registryCredentials) {
+        const { environment, ...restConfig } = userConfig;
+        fullConfig = {
+          environment: { registry_auth: registryCredentials, ...environment },
+          ...restConfig,
+        };
+      } else {
+        fullConfig = userConfig;
+      }
+
```
Comment (hkang1) [inline webui/react/src/hooks/useModal/useModalExperimentCreate.tsx:251]: **super nit suggestion:** this is pretty easy to read so feel free to ignore this. The logic can probably simplify a little, assuming the `SimpleMode` can follow the same exact logic as the `AdvancedMode` where you check to see if `environment?.registry_auth` is NOT defined and `registryCredentials` is to spot update `config > environment > registry_auth`.

File: webui/react/src/pages/ExperimentDetails/ExperimentConfiguration.tsx [hunk 2]
```diff
@@ -14,6 +14,15 @@ interface Props {
 const MonacoEditor = React.lazy(() => import('components/MonacoEditor'));
 
 const ExperimentConfiguration: React.FC<Props> = ({ experiment }: Props) => {
+  let publicConfig = {};
+  if (experiment.configRaw) {
+    const {
```
Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentConfiguration.tsx:19]: **suggestion:** could use a small comment here indicating that we are removing `registry_auth` from the config.

**Review by hkang1 — APPROVED** [review]
LGTM! I tried the test plan and everything seems to work nicely via netlify. I add credentials to experiments without them, removed credentials from ones that had them and changed credentials for ones that also had them defined. All worked seamlessly!

File: webui/react/src/pages/ExperimentDetails/ExperimentConfiguration.tsx (hunk 2 above)
Comment (trentwatt) [reply webui/react/src/pages/ExperimentDetails/ExperimentConfiguration.tsx:22]: 👍

File: webui/react/src/hooks/useModal/useModalExperimentCreate.tsx (hunk 1 above)
Comment (trentwatt) [reply webui/react/src/hooks/useModal/useModalExperimentCreate.tsx]: did something similar to simplify 👍

**Pull request merged**